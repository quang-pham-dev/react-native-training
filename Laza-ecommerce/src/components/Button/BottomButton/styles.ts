import { StyleSheet } from 'react-native';
// Styles
import { Metrics, Colors, Fonts } from '@styles/themes/index';

const styles = StyleSheet.create({
  button: {
    paddingVertical: Metrics.padding.md,
    paddingHorizontal: Metrics.padding.xl,
    alignItems: 'center',
  },
  title: {
    color: Colors.white,
    textTransform: 'none',
    fontFamily: Fonts.fontFamily.Inter_600SemiBold,
    fontSize: Fonts.size.sm,
    lineHeight: Fonts.lineHeight.lg,
  },
});

export default styles;
