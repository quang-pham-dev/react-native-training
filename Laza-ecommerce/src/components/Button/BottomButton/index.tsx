import React, { memo } from 'react';
import { Text, TouchableOpacity } from 'react-native';
// Types
import { BottomButtonProps } from 'types/Button';
// Styles
import styles from './styles';

const BottomButton = ({
  title,
  disabled,
  buttonStyles = [],
  children = [],
  onPress = () => {},
}: BottomButtonProps) => {
  return (
    <TouchableOpacity
      style={[styles.button, buttonStyles]}
      onPress={onPress}
      disabled={disabled}
      activeOpacity={0.8}
    >
      <Text style={[styles.title]}>{title}</Text>
      {children}
    </TouchableOpacity>
  );
};

export default memo(BottomButton);
