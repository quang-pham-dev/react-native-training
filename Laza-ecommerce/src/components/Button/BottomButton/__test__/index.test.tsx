import React from 'react';
import BottomButton from 'components/Button/BottomButton';
import renderer from 'react-test-renderer';

describe('Bottom Button Component', () => {
  test('should render correctly', () => {
    const tree = renderer
      .create(<BottomButton title='Create an Account' onPress={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render disabled button correctly', () => {
    const tree = renderer
      .create(<BottomButton title='Google' disabled onPress={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
