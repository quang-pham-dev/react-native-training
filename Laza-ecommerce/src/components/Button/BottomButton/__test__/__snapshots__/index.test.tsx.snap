// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bottom Button Component should render correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Inter_600SemiBold",
          "fontSize": 17,
          "lineHeight": 22,
          "textTransform": "none",
        },
      ]
    }
  >
    Create an Account
  </Text>
</View>
`;

exports[`Bottom Button Component should render disabled button correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 12,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#fff",
          "fontFamily": "Inter_600SemiBold",
          "fontSize": 17,
          "lineHeight": 22,
          "textTransform": "none",
        },
      ]
    }
  >
    Google
  </Text>
</View>
`;
